### PR TITLE
Create guideline for ASP .NET

### PR DESCRIPTION
**Pull request by aunsuwijak**

### CODE REVIEW

File: how-we-work/pages/asp-net-core-guideline.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+- MyProject.Api
+- MyProject.Domain
+- MyProject.Infrastructure
+- MyProject.Tests
+- MyProject.sln
+```
+
+### Infrastructure
+
+As you can see above there is a project call `MyProject.Api`. This project store all the code related to database, storage, etc. such as `DbContext`, `DbSet` and `Repository`
```
Comment (PeeraJ) [inline how-we-work/pages/asp-net-core-guideline.md]: the project name should be `MyProject.Infrastructure`

File: how-we-work/pages/asp-net-core-guideline.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+### Domain
+
+As you can see above there is a project call `MyProject.Api`. This project store all the code related to application logic such as `Model`, `Service`, etc.
```
Comment (PeeraJ) [inline how-we-work/pages/asp-net-core-guideline.md]: the project name should be `MyProject.Domain`

File: how-we-work/pages/asp-net-core-guideline.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
+# ASP .NET Development Guideline
+
+One of AppMan main backend technology is ASP .NET. Here is the guideline of how our team develop ASP .NET application.
+
+## Project Structure
+
+Normally, we create 3 projects when we start developing any ASP .NET solution.
```
Comment (whai-kung) [inline how-we-work/pages/asp-net-core-guideline.md]: 4 projects

File: how-we-work/pages/asp-net-core-guideline.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+- MyProject.sln
+```
+
+### Infrastructure
+
+As you can see above there is a project call `MyProject.Infrastructure`. This project store all the code related to database, storage, etc. such as `DbContext`, `DbSet` and `Repository`
+
+### Domain
+
+As you can see above there is a project call `MyProject.Domain`. This project store all the code related to application logic such as `Model`, `Service`, etc.
```
Comment (whai-kung) [inline how-we-work/pages/asp-net-core-guideline.md]: there is a project `called`

Comment (SumetKS) [inline how-we-work/pages/asp-net-core-guideline.md]: And business logic in here, right?

Comment (SumetKS) [inline how-we-work/pages/asp-net-core-guideline.md]: describe about 'Service'

File: how-we-work/pages/asp-net-core-guideline.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+- Create http client using `WebApplicationFactory<Startup>` and `CreateClient` method.
+- Use http client to create request to controller that you want to test and assign the response to variable.
+- Verify the response by `xUnit` assertion.
+- Integration test will have only 1 class per file which different from unit test.
+- There will be a prefix of http method such as `Get_`, `Post_`, etc. in front of each test case.
+- Use `AAA` pattern the same as unit test.
+
+### Test Coverage
+
+To be able to check how many test we need to write we will use `coverlet` and `dotnet-reportgenerator-cli` to generate `opencover` HTML test coverage report
```
Comment (whai-kung) [inline how-we-work/pages/asp-net-core-guideline.md]: please add the reference to how to install and run test coverlet.

File: how-we-work/pages/asp-net-core-guideline.md [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+  - SQLMigrations
+    - Down_20200103080406_CreateTweet.cs
+    - Up_20200103080406_CreateTweet.cs
+```
+
+We will send our user the script in `SQLMigrations` folder to customer.
+
+### Mock
+
+We create mock service for all 3rd party services with `API Blueprint` which we can get both document and mock serivces.
```
Comment (whai-kung) [inline how-we-work/pages/asp-net-core-guideline.md]: do we need api blueprint for ourselves?

File: how-we-work/pages/asp-net-core-guideline.md [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+
+```markdown
+- documents
+  - api.apib
+  - package.json
```
Comment (whai-kung) [inline how-we-work/pages/asp-net-core-guideline.md]: where should we put our document specs such as data dic and stuff?

File: how-we-work/pages/asp-net-core-guideline.md [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+{
+    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public int Id { get; set; }
+    public string Name { get; set; }
+}
+```
+
+#### Create & Apply Migration
+
+Now it's time to create database migration from our existing code. Use keyword `Add-Migration` in powershell.
```
Comment (SumetKS) [inline how-we-work/pages/asp-net-core-guideline.md]: 'Add-Migration' is valid on 'Package Manager Console' in Visual Studio only if you have to use .NET Core CLI
dotnet ef migrations add xxxxx

File: how-we-work/pages/asp-net-core-guideline.md (hunk 7 above)
Comment (krihsana) [inline how-we-work/pages/asp-net-core-guideline.md]: In section documents should have sequence diagram and ER diagram ?

**Pull request merged**